File: ex189.py
class Solution:
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: void Do not return anything, modify nums in-place instead.
        """

        for i in range(k):  nums.insert(0, nums.pop(-1))

        return nums

sol = Solution()
print(sol.rotate([-1, -100, 3, 99], 2))

File: code/ex189.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        reverse(nums.begin(), nums.end());    // Reverse entire
        reverse(nums.begin(), nums.begin() + k);    // Reverse front k
        reverse(nums.begin() + k, nums.end());    // Reverse last n - k
    }
};